nixos/modules/services/matrix/mautrix-signal.nix

```diff
@@ -0,0 +1,208 @@
+  settingsFormat = pkgs.formats.json { };
+  appservicePort = 29328;
+
+  mkDefaults = lib.mapAttrsRecursive (n: v: lib.mkDefault v);
+  defaultConfig = {
+    homeserver.address = "http://localhost:8448";
+    appservice = {
+      hostname = "[::]";
+      port = appservicePort;
+      database.type = "sqlite3";
+      database.uri = "file:${dataDir}/mautrix-signal.db?_txlock=immediate";
+      id = "signal";
+      bot = {
+        username = "signalbot";
+        displayname = "Signal Bridge Bot";
+      };
+      as_token = "";
+      hs_token = "";
+    };
+    bridge = {
+      username_template = "signal_{{.}}";
+      displayname_template = "{{or .ProfileName .PhoneNumber \"Unknown user\"}}";
+      double_puppet_server_map = { };
+      login_shared_secret_map = { };
+      command_prefix = "!signal";
+      permissions."*" = "relay";
+      relay.enabled = true;
+    };
+    logging = {
+      min_level = "info";
+      writers = lib.singleton {
+        type = "stdout";
+        format = "pretty-colored";
+        time_format = " ";
+      };
+    };
+  };
+
+in
+{
+  options.services.mautrix-signal = {
+    enable = lib.mkEnableOption (lib.mdDoc "mautrix-signal, a puppeting/relaybot bridge between Matrix and Signal.");
+
+    settings = lib.mkOption {
+      type = settingsFormat.type;
+      default = defaultConfig;
+      description = lib.mdDoc ''
+        {file}`config.yaml` configuration as a Nix attribute set.
+        Configuration options should match those described in
+        [example-config.yaml](https://github.com/mautrix/signal/blob/master/example-config.yaml).
+        Secret tokens should be specified using {option}`environmentFile`
+        instead of this world-readable attribute set.
+      '';
+      example = {
+        appservice = {
+          database = {
+            type = "postgres";
+            uri = "postgresql:///mautrix_signal?host=/run/postgresql";
+          };
+          id = "signal";
+          ephemeral_events = false;
+        };
+        bridge = {
+          history_sync = {
+            request_full_sync = true;
+          };
+          private_chat_portal_meta = true;
+          mute_bridging = true;
+          encryption = {
+            allow = true;
+            default = true;
+            require = true;
+          };
+          provisioning = {
+            shared_secret = "disable";
+          };
+          permissions = {
+            "example.com" = "user";
+          };
+        };
+      };
+    };
+    environmentFile = lib.mkOption {
+      type = lib.types.nullOr lib.types.path;
+      default = null;
+      description = lib.mdDoc ''
+        File containing environment variables to be passed to the mautrix-signal service,
+        in which secret tokens can be specified securely by optionally defining a value for
+        `MAUTRIX_SIGNAL_BRIDGE_LOGIN_SHARED_SECRET`.
```

Comment:
I cannot find any documentation on this environment variable, can you add a pointer to its usage?

Reply:
I double checked myself and I think you are correct. The [docs](https://docs.mau.fi/bridges/general/double-puppeting.html) only reference the YAML way of configuring this, which is also what I'm using on my own server. I originally copied this over from the mautrix-whatsapp module, but I ran into the same issue there and had to put the secret into the config attribute set instead. Maybe @frederictobiasc (the maintainer of the mautrix-whatsapp module) can shine some light on this?

Reply:
To be more precise, I think this environment variable has been removed in both mautrix-signal and mautrix-whatsapp when they were rewritten in Go.

Reply:
@Ma27 Are you using double puppeting with your Signal bridge? If so, how do you configure it?

Reply:
> To be more precise, I think this environment variable has been removed in both mautrix-signal and mautrix-whatsapp when they were rewritten in Go.

Never mind that, sorry, it's been awhile since setting this up. :) The service file is using envsubst to generate the file configuration, we only have to adjust the name and format to fit the new secret "map", which is needed because the secrets are now configured per domain. I'll look into it.